<?xml version="1.0"?>
<sdf version="1.9">

  <model name="clarke">

    <!-- Body -->
    <link name="base_link">

      <!-- For more information about how to calculate the inertia values: 
      https://en.wikipedia.org/wiki/List_of_moments_of_inertia#List_of_3D_inertia_tensors -->
      <!-- https://classic.gazebosim.org/tutorials?tut=inertia&cat=build_robot -->

      <pose>0 0 0 1.5708 1.5708 -1.5708</pose>
      <!-- <pose>0 0 0 1.5708 0 0</pose> -->

      <!-- <inertial>
        <mass>147.8671</mass>
        <inertia>
          <ixx>3.000000</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>41.980233</iyy>
          <iyz>0</iyz>
          <izz>41.980233</izz>
        </inertia>
      </inertial> -->

      <inertial>
        <mass>22.3</mass>
        <inertia>
          <ixx>0.1376267915</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.6490918050833332</iyy>
          <iyz>0</iyz>
          <izz>0.6490918050833332</izz>
        </inertia>
      </inertial> 
        

      <collision name="main_body_collision">
        <pose>0 0 0 1.5708 1.5708 1.5708</pose>
        <geometry>
          <cylinder>
            <radius>0.1111</radius>
            <length>0.5588</length>
          </cylinder>
        </geometry>
      </collision>

      <visual name="visual">
        <geometry>
          <cylinder>
            <radius>0.1111</radius>
            <length>0.5588</length>
          </cylinder>
          <!-- <mesh>
            <uri>meshes/clarke.dae</uri>
          </mesh> -->
        </geometry>
      </visual>

      <!-- IMU sensor -->
      <sensor name="imu_sensor" type="imu">
        <always_on>1</always_on>
        <update_rate>7</update_rate>
        <visualize>true</visualize>
        <topic>imu</topic>
      </sensor>

      <!-- Down camera -->
      <sensor name="down_cam" type="camera">
        <pose>0 0 0 3.1415 0 0</pose>
        <horizontal_fov>3.83</horizontal_fov> 
        <update_rate>10.0</update_rate>
        <visualize>true</visualize>
        <topic>vision/down_cam/image_raw</topic>
        <always_on>1</always_on>
        <image>
          <width>1280</width>
          <height>962</height>
        </image>
      </sensor>

      <!-- Depth camera -->
      <sensor name="front_cam_depth" type="depth_camera">
        <pose>0.1 0 -0.05 3.13159 1.5708 0</pose>
        <always_on>1</always_on>
        <visualize>true</visualize>
        <topic>vision/front_cam/depth</topic>
        <horizontal_fov>1.57</horizontal_fov>
        <update_rate>10.0</update_rate>
        <image>
          <width>512</width>
          <height>370</height>
          <format>B8G8R8</format>
        </image>
      </sensor>

      <!-- Front -->
      <sensor name="front_cam_rgb" type="camera">
        <pose>0.1 0 -0.05 3.13159 1.5708 0</pose>
        <camera>
          <horizontal_fov>1.57</horizontal_fov>
          <image>
            <width>512</width>
            <height>370</height>
          </image>
        </camera>
        <always_on>1</always_on>
        <update_rate>10</update_rate>
        <visualize>true</visualize>
        <topic>vision/front_cam/image_rgb</topic>
      </sensor>

    </link>

    <!-- Thruster0 -->
    <link name="thruster0">
      <pose relative_to="base_link">0 -0.2127 0 0 1.5708 0</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>    

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision> 
      
      <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual>

    </link>

    <!-- Thruster1 -->
    <link name="thruster1">  
      <pose relative_to="base_link">0 0.2127 0 0 1.5708 0</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>    

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision> 

      <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
            
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>

      </visual>

    </link>

    <!-- Thruster2 -->
    <link name="thruster2">
      <pose relative_to="base_link">0 -0.1111 -0.4191 0 0 -1.5708</pose> 

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>    

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision> 

      <!-- <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
            
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual> -->

    </link>

    <!-- Thruster3 -->
    <link name="thruster3">      
      <pose relative_to="base_link">0 0.1111 0.4191 0 0 1.5708</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision>

      <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
            
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual>

    </link>

    <!-- Thruster4 -->
    <link name="thruster4">
      <pose relative_to="base_link">0 -0.2127 0.2794 0 3.1415 0</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision>

      <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual>

    </link>

    <!--Thruster5 -->
    <link name="thruster5">
      <pose relative_to="base_link">0 0.2127 0.2794 0 3.1415 0</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision>

      <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual>

    </link>

    <!--Thruster6 -->
    <link name="thruster6">
      <pose relative_to="base_link">0 0.2127 -0.2794 0 3.1415 0</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>    

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision> 

      <!-- <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual> -->

    </link>

    <!--Thruster7 -->
    <link name="thruster7">      
      <pose relative_to="base_link">0 -0.2127 -0.2794 0 3.1415 0</pose>

      <inertial>
        <mass>0.2</mass>
        <inertia>
          <ixx>0.000143971303</ixx>
          <ixy>0.000000000008</ixy>
          <ixz>-0.000000000224</ixz>
          <iyy>0.000140915448</iyy>
          <iyz>-0.000025236433</iyz>
          <izz>0.000033571862</izz>
        </inertia>
      </inertial>  

      <collision name="collision">
        <geometry>
          <box>
            <size>0.03 0.1 0.03</size>
          </box>
        </geometry>
      </collision>   

      <!-- <visual name= "visual">
        <geometry>
          <mesh>
            <uri>model://assets/clarke/meshes/tethys.dae</uri>
            <submesh>
              <name>Prop</name>
              <center>true</center>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <diffuse>1.0 1.0 1.0</diffuse>
          <specular>1.0 1.0 1.0</specular>
          <pbr>
            <metal>
            
              <albedo_map>model://assets/clarke/materials/textures/Tethys_Albedo.png</albedo_map>
              <normal_map>model://assets/clarke/materials/textures/Tethys_Normal.png</normal_map>
              <metalness_map>model://assets/clarke/materials/textures/Tethys_Metalness.png</metalness_map>
              <roughness_map>model://assets/clarke/materials/textures/Tethys_Roughness.png</roughness_map>
            </metal>
          </pbr>
          <script>
            <uri>model://assets/clarke/materials/scripts/</uri>
            <uri>model://assets/clarke/materials/textures/</uri>
            <name>model://assets/clarke/tethys/Tethys_Diffuse</name>
          </script>
        </material>
      </visual> -->

    </link>

    <!-- Joints -->
    <joint name="thruster0_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster0</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster1_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster1</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster2_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster2</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster3_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster3</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster4_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster4</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster5_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster5</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster6_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster6</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <joint name="thruster7_joint" type="revolute">
      <parent>base_link</parent>
      <child>thruster7</child>
      <axis>
        <xyz>1 0 0</xyz>
        <limit>
          <lower>-1e+12</lower>
          <upper>1e+12</upper>
          <effort>-1</effort>
          <velocity>-1</velocity>
        </limit>
      </axis>
    </joint>

    <plugin 
        filename="libignition-gazebo-imu-system.so"
        name="ignition::gazebo::systems::Imu">
    </plugin>

    <plugin
        filename="libignition-gazebo-sensors-system.so"
        name="ignition::gazebo::systems::Sensors">
        <render_engine>ogre</render_engine>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster0_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster1_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster2_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster3_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster4_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster5_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster6_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>

    <plugin
        filename="ignition-gazebo-thruster-system"
        name="ignition::gazebo::systems::Thruster">
        <namespace>clarke</namespace>
        <joint_name>thruster7_joint</joint_name>
        <thrust_coefficient>0.004422</thrust_coefficient>
        <fluid_density>997</fluid_density>
        <propeller_diameter>0.09</propeller_diameter>
    </plugin>
    
    <!-- <plugin
        filename="ignition-gazebo-hydrodynamics-system"
        name="ignition::gazebo::systems::Hydrodynamics">
        <link_name>base_link</link_name>
        <xDotU>-4.876161</xDotU>
        <yDotV>-126.324739</yDotV>
        <zDotW>-126.324739</zDotW>
        <kDotP>0</kDotP>
        <mDotQ>-33.46</mDotQ>
        <nDotR>-33.46</nDotR>
        <xUU>-6.2282</xUU>
        <xU>0</xU>
        <yVV>-601.27</yVV>
        <yV>0</yV>
        <zWW>-601.27</zWW>
        <zW>0</zW>
        <kPP>-0.1916</kPP>
        <kP>0</kP>
        <mQQ>-632.698957</mQQ>
        <mQ>0</mQ>
        <nRR>-632.698957</nRR>
        <nR>0</nR>
    </plugin>  -->

    <!-- 
      Xdotu = -2.171640047
      Ydotv = Zdotw = -20.12043854
      Kdotp = 0
      Ndotr = Ndotq = -0.898652566
     -->


    <!-- <plugin
        filename="ignition-gazebo-hydrodynamics-system"
        name="ignition::gazebo::systems::Hydrodynamics">
        <link_name>base_link</link_name>
        <xDotU>-4.876161</xDotU>
        <yDotV>-126.324739</yDotV>
        <zDotW>-126.324739</zDotW>
        <kDotP>0</kDotP>
        <mDotQ>-33.46</mDotQ>
        <nDotR>-33.46</nDotR>
        <xUU>0</xUU>
        <xU>0</xU>
        <yVV>0</yVV>
        <yV>0</yV>
        <zWW>0</zWW>
        <zW>0</zW>
        <kPP>0</kPP>
        <kP>0</kP>
        <mQQ>0</mQQ>
        <mQ>0</mQ>
        <nRR>0</nRR>
        <nR>0</nR>
    </plugin>      -->

    <!-- <plugin
          filename="ignition-gazebo-lift-drag-system"
          name="ignition::gazebo::systems::LiftDrag">
              <air_density>1000</air_density>
              <cla>4.13</cla>
              <cla_stall>-1.1</cla_stall>
              <cda>0.2</cda>
              <cda_stall>0.03</cda_stall>
              <alpha_stall>0.17</alpha_stall>
              <a0>0</a0>
              <area>0.0244</area>
              <upward>0 1 0</upward>
              <forward>1 0 0</forward>
              <link_name>vertical_fins</link_name>
              <cp>0 0 0</cp>
          </plugin> -->

    <plugin
        filename="ignition-gazebo-hydrodynamics-system"
        name="ignition::gazebo::systems::Hydrodynamics">
        <link_name>base_link</link_name>
        <xDotU>-5.171840284722776</xDotU>
        <yDotV>-15.233882678825832</yDotV>
        <zDotW>-15.233882678825832</zDotW>
        <kDotP>0</kDotP>
        <mDotQ>-0.09103716131747328</mDotQ>
        <nDotR>-0.09103716131747328</nDotR>
        <xUU>-15.898708521663591</xUU>
        <xU>0</xU>
        <yVV>-50.9077976</yVV>
        <yV>0</yV>
        <zWW>-50.9077976</zWW>
        <zW>0</zW>
        <kPP>-0</kPP>
        <kP>-0</kP>
        <mQQ>-0.06981142220562829</mQQ>
        <mQ>0</mQ>
        <nRR>-0.06981142220562829</nRR>
        <nR>0</nR>
    </plugin> 

  </model>
</sdf>
